runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -921,20 +925,436 @@ static void VMwrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register
+   loadAddressConstant(cg, node, reinterpret_cast<uintptrj_t>(methodSymbol->getResolvedMethod()->constantPool()), resultReg);
+   loadConstant(cg, node, symRef->getCPIndex(), cpIndexReg);
+
+   // cpAddress is the first argument of VMHelper
+   deps->addPreCondition(resultReg, TR::RealRegister::gr3);
+   deps->addPostCondition(resultReg, TR::RealRegister::gr3);
+   // cpIndexReg is the second argument
+   deps->addPreCondition(cpIndexReg, TR::RealRegister::gr4);
+   deps->addPostCondition(cpIndexReg, TR::RealRegister::gr4);
+
+   // Generate helper address and branch
+   TR::SymbolReference *helperSym = comp->getSymRefTab()->findOrCreateRuntimeHelper(helperIndex, false, false, false);
+   TR::Instruction *call = generateDepImmSymInstruction(cg, TR::InstOpCode::bl, node, reinterpret_cast<uintptrj_t>(helperSym->getMethodAddress()), deps, helperSym);
+   call->PPCNeedsGCMap(linkageProperties.getPreservedRegisterMapForGC());
+
+   /*
+      * For instance field offset, the result returned by the vmhelper includes header size.
+      * subtract the header size to get the offset needed by field watch helpers
+   */
+   if (!isStatic)
+      addConstantToInteger(node, resultReg, resultReg , -TR::Compiler->om.objectHeaderSizeInBytes(), cg);
+
+   // store result into J9JITWatchedStaticFieldData.fieldAddress / J9JITWatchedInstanceFieldData.offset
+   TR::MemoryReference* dataRef = new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetInDataBlock, TR::Compiler->om.sizeofReferenceAddress(), cg);
+   generateMemSrc1Instruction(cg, TR::InstOpCode::stw, node, dataRef, resultReg);
+   
+   generateLabelInstruction(cg, TR::InstOpCode::b, node, endLabel);
+      
+   generateReportOOL->swapInstructionListsWithCompilation();
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, endLabel);
+
+   cg->stopUsingRegister(scratchReg);
+   cg->stopUsingRegister(cndReg);
+   cg->stopUsingRegister(cpIndexReg);
+   cg->stopUsingRegister(resultReg);
+
+   }
+
+/*
+ * Generate the reporting field access helper call with required arguments
+ *
+ * jitReportInstanceFieldRead
+ * arg1 pointer to static data block
+ * arg2 object being read
+ *
+ * jitReportInstanceFieldWrite
+ * arg1 pointer to static data block
+ * arg2 object being written to
+ * arg3 pointer to value being written
+ *
+ * jitReportStaticFieldRead
+ * arg1 pointer to static data block
+ *
+ * jitReportStaticFieldWrite
+ * arg1 pointer to static data block
+ * arg2 pointer to value being written
+ *
+ */
+void generateReportFieldAccessOutlinedInstructions(TR::Node *node, TR::LabelSymbol *endLabel, TR::Register *dataBlockReg, bool isWrite, TR::RegisterDependencyConditions *deps, TR::CodeGenerator *cg, TR::Register *sideEffectRegister, TR::Register *valueReg)
+   {
+   bool isInstanceField = node->getOpCode().isIndirect();
+   J9Method *owningMethod = reinterpret_cast<J9Method *>(node->getOwningMethod());
```

Comment:
This is not used anywhere. So we can delete it.

Reply:
Thanks. 
See https://github.com/eclipse/openj9/pull/6234/commits/a5abb8d230c501f681024349304c2305379d94e5